pageserver/src/layered_repository/storage_layer.rs

```diff
@@ -123,6 +123,14 @@ pub trait Layer: Send + Sync {
         reconstruct_data: &mut ValueReconstructState,
     ) -> Result<ValueReconstructResult>;
 
+    /// Find latest LSN of wal record matching specified criteria
+    fn find(
+        &self,
+        key: Key,
+        lsn: Lsn,
+        filter: &dyn Fn(ZenithWalRecord) -> bool,
+    ) -> Result<Option<Lsn>>;
```

Comment:
it seems like a generic name for very specific function.
also should it better return a log record as well (so that we can extract a timestamp too)?

Reply:
I renamed function to `find_record_lst`.
Concerning your proposal to return located record: right now it is not needed, because we actually perform lookup by timestamp/xid.
If sometimes we will need it, there is no problem to change this function type to return pair <Lsn,ZenithWalRecord>. But right now I prefer to avoid redundant memory copies and do not return record until it is really needed.